lib/internal/fs/promises.js

```diff
@@ -808,11 +808,27 @@ async function readFile(path, options) {
   return handleFdClose(readFileHandle(fd, options), fd.close);
 }
 
+async function exists(path) {
+  path = getValidatedPath(path, 'path');
+  try {
+    await access(path, F_OK);
+    return true;
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      return false;
+    }
+    // We are unable to determine if the file exists or not,
+    // like encountering a permission error.
+    throw err;
```

Comment:
Should always (including the fact that this is a permission error) return `false` in this `catch` block, and should never throw an error to mimic `existsSync` API.